### PR TITLE
Retry reset before request integration

### PR DESCRIPTION
**Pull request by keithmattix**
**Please provide a description of this PR:**
Add validations and tests for reset-before-request retry policy envoyproxy/envoy#35074

### CODE REVIEW

**Comment by keithmattix** [comment]
Fixes #51704

**Review by bleggett — APPROVED** [review]

File: pkg/config/validation/validation.go [hunk 1]
```diff
@@ -84,6 +84,7 @@ var (
 		"5xx",
 		"gateway-error",
 		"reset",
+		"reset-before-request",
```
Comment (hzxuzhonghu) [inline pkg/config/validation/validation.go:87]: should check the proxy version before applying to it

Comment (keithmattix) [reply pkg/config/validation/validation.go:87]: This is for the VirtualService validation right? IIUC, that's not proxy dependent. If there's even 1 proxy with the latest version, it's valid. Or is your point that if there's a single proxy on an old version, that proxy's RDS will get NACK'd? L

Comment (ramaraochavali) [reply pkg/config/validation/validation.go:87]: If set this after Istio upgrade and no proxy restart (proxy running old version) - wouldn't this cause RDS NACK?

Comment (ramaraochavali) [reply pkg/config/validation/validation.go:87]: just put a hold to resolve this comment

Comment (keithmattix) [reply pkg/config/validation/validation.go:87]: If a user sets it yes; all this does is not block a VS setting this new value. Adding a version gate here stops user error but how would we even write a validation to check all of the proxy versions and conditionally allow the VS?

Comment (ramaraochavali) [reply pkg/config/validation/validation.go:87]: > Adding a version gate here stops user error but how would we even write a validation to check all of the proxy versions and conditionally allow the VS?

Yeah. I wasn't suggesting to add validation check. Just want to understand how it works. Just looked at the Envoy [code](https://github.com/envoyproxy/envoy/blob/main/source/common/router/retry_state_impl.cc#L190)  - It does not NACK. It uses the validation only in the strict header check mode when retry on is passed via x-envoy-retry-on header. So we are good here

Comment (howardjohn) [reply pkg/config/validation/validation.go:87]: If it did NACK, the fix IMO should be to silently remove it for old proxies if anything. Good to know nothing is needed though.

validation.go cannot be proxy dependant though

Comment (hzxuzhonghu) [reply pkg/config/validation/validation.go:87]: But this would fail older proxy connecting with new isitod

Comment (ramaraochavali) [reply pkg/config/validation/validation.go:87]: Why would it fail?

**Pull request merged**